ping/ping.c

```diff
@@ -647,7 +647,7 @@ main(int argc, char **argv)
 	int target_ai_family = hints.ai_family;
 	hints.ai_family = AF_UNSPEC;
 
-	unsigned char buf[sizeof(struct in6_addr)];
+	char buf[INET6_ADDRSTRLEN];
```

Comment:
I don't think this is correct. buf used for inet_pton() will contain a binary address structure. It's not related to  INET6_ADDRSTRLEN. inet_pton(3) documents the needed buffer size and I think it's already correct before this patch.

Reply:
Thanks a lot, I'll remove the patch (I only move the variable declaration so that it can be reused).

Reply:
@gobenji  FYI `sizeof(struct in6_addr)` is `16`, which is the same as `INET_ADDRSTRLEN`. `INET6_ADDRSTRLEN` is `46`, which should be for really strange addresses, e.g. IPv4-mapped formIPv4-mapped form `ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255`, but it was working with original form, thus I remove this commit to speedup the review.